provider/aws/aws_discover.go

```diff
@@ -133,33 +135,36 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 	l.Printf("[INFO] discover-aws: Region is %s", region)
 
 	l.Printf("[DEBUG] discover-aws: Creating session...")
-	config := aws.Config{
-		Region: &region,
-		Credentials: credentials.NewChainCredentials(
-			[]credentials.Provider{
-				&credentials.StaticProvider{
-					Value: credentials.Value{
-						AccessKeyID:     accessKey,
-						SecretAccessKey: secretKey,
-						SessionToken:    sessionToken,
-					},
-				},
-				&credentials.EnvProvider{},
-				&credentials.SharedCredentialsProvider{},
-				defaults.RemoteCredProvider(*(defaults.Config()), defaults.Handlers()),
-			}),
-	}
-	if endpoint != "" {
-		l.Printf("[INFO] discover-aws: Endpoint is %s", endpoint)
-		config.Endpoint = &endpoint
+	var cfg aws.Config
+	var err error
+	if accessKey != "" && secretKey != "" {
+		log.Println("Using static credentials provider")
+		staticCreds := credentials.NewStaticCredentialsProvider(accessKey, secretKey, sessionToken)
+		cfg, err = config.LoadDefaultConfig(context.TODO(), config.WithRegion(region),
+			config.WithCredentialsProvider(aws.NewCredentialsCache(staticCreds)))
+		if err != nil {
+			l.Printf("unable to load SDK config with Static Provider, %v", err)
+		}
+	} else {
+		log.Println("Using default credential chain")
+		cfg, err = config.LoadDefaultConfig(context.TODO(),
+			config.WithRegion(region), // Specify your region
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load SDK config with default credential chain, %s", err)
+		}
 	}
 
 	// Split here for ec2 vs ecs decision tree
 	if service == "ecs" {
-		svc := ecs.New(session.New(), &config)
+		svc := ecs.NewFromConfig(cfg, func(o *ecs.Options) {
+			if endpoint != "" {
+				o.BaseEndpoint = aws.String(endpoint)
```

Comment:
A high level observation, the logging is degraded.

 ```suggestion
				o.BaseEndpoint = aws.String(endpoint)
				l.Printf("[INFO] discover-aws: Endpoint is %s", endpoint)
```

Reply:
Updated the log statements.